lib/internal/modules/esm/default_resolve.js

```diff
@@ -93,7 +101,24 @@ function resolve(specifier, parentURL) {
       throw new ERR_INPUT_TYPE_NOT_ALLOWED();
   }
 
-  let url = moduleWrapResolve(specifier, parentURL);
+  let url;
+  try {
+    url = moduleWrapResolve(specifier, parentURL);
+  } catch (error) {
+    error.stack;
+    let e = error;
+    try {
+      debugger
+      const fakeParentUrl = new url.URL(parentURL)
+      const fakeParent = new Module(fakeParentUrl.pathname, null)
```

Comment:
@guybedford When I try to create a "fake module" of the parent, that module won't have `require` function, so, anyway will trows `ERR_MODULE_NOT_FOUND`, how could I build a "good" module?

`Module._compile(...)` seems to take care of that setup, I'm pretty lost with this

Reply:
I saw [this comment](https://github.com/nodejs/node/issues/30645#issuecomment-558366231), may it could help us with this.

Reply:
Here's how it used to be done (from https://github.com/nodejs/node/pull/14369/files#diff-3e94629c67625a2547d1507d0a547211):

```js
module.exports = (target, base) => {
  target = `${target}`;
  if (base === undefined) {
    // We cannot search without a base.
    throw new errors.Error('ERR_MISSING_MODULE', target);
  }
  base = `${base}`;
  try {
    return resolve(target, base);
  } catch (e) {
    e.stack; // cause V8 to generate stack before rethrow
    let error = e;
    try {
      const questionedBase = new URL(base);
      const tmpMod = new CJSmodule(questionedBase.pathname, null);
      tmpMod.paths = CJSmodule._nodeModulePaths(
        new URL('./', questionedBase).pathname);
      const found = CJSmodule._resolveFilename(target, tmpMod);
      error = new errors.Error('ERR_MODULE_RESOLUTION_LEGACY', target,
                               base, found);
    } catch (problemChecking) {
      // ignore
    }
    throw error;
  }
};
```

Reply:
Thanks! I could missed something because of being rushed or tired! Now it's a bit more clear.

Reply:
Marking this comment as resolved. Just let me know if you need further feedback at all.